src/intrinsics/Iterator.mts

```diff
@@ -31,8 +37,33 @@ function* IteratorConstructor(
   return Q(yield* OrdinaryCreateFromConstructor(NewTarget, '%Iterator.prototype%'));
 }
 
+/** https://tc39.es/ecma262/#sec-iterator.from */
+function* Iterator_from([O]: Arguments): ValueEvaluator {
+  // 1. Let iteratorRecord be ? GetIteratorFlattenable(O, iterate-string-primitives).
+  const iteratorRecord = Q(yield* GetIteratorFlattenable(O, 'iterate-string-primitives'));
+
+  // 2. Let hasInstance be ? OrdinaryHasInstance(%Iterator%, iteratorRecord.[[Iterator]]).
+  const hasInstance = Q(yield* OrdinaryHasInstance(surroundingAgent.intrinsic('%Iterator%'), iteratorRecord.Iterator));
+  // 3. If hasInstance is true, then
+  if (hasInstance.booleanValue()) {
```

Comment:
in the codebase it usually should be `hasInstance === Value.true`, but this is not wrong either

Reply:
fixed